apps/dgst.c

```diff
@@ -413,6 +413,28 @@ int dgst_main(int argc, char **argv)
     return ret;
 }
 
+const char * newline_escape_filename(const char *file, int * backslash)
+{
+    int i = 0, length = strlen(file);
+    char *file_cpy = app_malloc(length + 3, file);
+    while(i < length) {
+        const char c = file[i];
+        if (c == '\n') {
+            file_cpy[i] = '\\';
+            i += 1;
+            file_cpy[i] = 'n';
+            i += 1;
+            *backslash = 1;
+        } else {
+            file_cpy[i] = c;
+            i += 1;
+        }
+    }
+    file_cpy[i] = '\0';
+    return (const char *)file_cpy;
```

Comment:
How about having the function return `char *` and avoid this cast and the one in the free?

Reply:
This is a good point.  I was unsure if removing `const` from the `char *` would be desired since that is how it entered `do_fp`, but I can certainly do this.  Any final thoughts?

Reply:
I received a warning when discarding `const` from file and freeing it using `OPENSSL_free`
```
apps/dgst.c: In function 'do_fp':
apps/dgst.c:512:22: warning: passing argument 1 of 'CRYPTO_free' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
         OPENSSL_free(file);
include/openssl/crypto.h:129:21: note: in definition of macro 'OPENSSL_free'
         CRYPTO_free(addr, OPENSSL_FILE, OPENSSL_LINE)
```